### PR TITLE
[CPU Plugin][Functional Test] upgrade infer request related to 2.0

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Add infer request config test
 - Remove 1.0 properties with 2.0
 - *...*

### Tickets:
 - *CVS-128365*

### CODE REVIEW

File: src/tests/functional/plugin/shared/include/behavior/ov_infer_request/properties_tests.hpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+class InferRequestPropertiesTest : public testing::WithParamInterface<InferRequestPropertiesParams>,
+                                   public ov::test::behavior::OVInferRequestTestBase {
+public:
+    void SetUp() override {
+        std::tie(streamExecutorNumber, target_device, configuration) = this->GetParam();
+        // Skip test according to plugin specific disabledTestPatterns() (if any)
+        SKIP_IF_CURRENT_TEST_IS_DISABLED()
+        APIBaseTest::SetUp();
+        // Create CNNNetwork from ngrpah::Function
```
Comment (riverlijunjie) [inline src/tests/functional/plugin/shared/include/behavior/ov_infer_request/properties_tests.hpp:28]: Delete CNNNetwork.

File: src/tests/functional/plugin/shared/include/behavior/ov_infer_request/properties_tests.hpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+protected:
+    ov::CompiledModel execNet;
+    std::shared_ptr<ov::Core> core = ov::test::utils::PluginCache::get().core();
+    std::shared_ptr<ov::Model> function;
+    ov::AnyMap configuration;
+    size_t streamExecutorNumber;
+
+    void set_api_entity() override {
+        api_entity = ov::test::utils::ov_entity::ie_infer_request;
```
Comment (riverlijunjie) [inline src/tests/functional/plugin/shared/include/behavior/ov_infer_request/properties_tests.hpp:65]:
```suggestion
        api_entity = ov::test::utils::ov_entity::ov_infer_request;
```

**Review by riverlijunjie — APPROVED** [review]
LGTM

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/infer_request/memory_states.cpp [hunk 3]
```diff
@@ -1,24 +0,0 @@
...
-#include <common_test_utils/test_constants.hpp>
-#include "behavior/infer_request/memory_states.hpp"
-#include "functional_test_utils/plugin_cache.hpp"
-#include "ov_models/builders.hpp"
-
-using namespace BehaviorTestsDefinitions;
-
-namespace {
-std::vector<memoryStateParams> memoryStateTestCases = {
-    memoryStateParams(InferRequestVariableStateTest::getNetwork(), {"c_1-3", "r_1-3"}, ov::test::utils::DEVICE_CPU, {}),
```
Comment (vurusovs) [inline src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/infer_request/memory_states.cpp:14]: What is an equivalent for the test in API2.0?

Comment (zhaixuejun1993) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/infer_request/memory_states.cpp:14]: This PR only for memory_states: https://github.com/openvinotoolkit/openvino/pull/21627

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/infer_request/config.cpp [hunk 4]
```diff
@@ -1,28 +0,0 @@
...
-    };
-
-    const std::vector<std::map<std::string, std::string>> InConfigs = {
-            {},
-            {{InferenceEngine::PluginConfigParams::KEY_CPU_THROUGHPUT_STREAMS, InferenceEngine::PluginConfigParams::CPU_THROUGHPUT_AUTO}},
-            {{InferenceEngine::PluginConfigParams::KEY_CPU_THROUGHPUT_STREAMS, InferenceEngine::PluginConfigParams::CPU_THROUGHPUT_NUMA}},
-            {{InferenceEngine::PluginConfigParams::KEY_CPU_THROUGHPUT_STREAMS, "8"}},
-            {{InferenceEngine::PluginConfigParams::KEY_CPU_BIND_THREAD, InferenceEngine::PluginConfigParams::NO}},
-            {{InferenceEngine::PluginConfigParams::KEY_CPU_BIND_THREAD, InferenceEngine::PluginConfigParams::YES}}
-    };
```
Comment (vurusovs) [inline src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/infer_request/config.cpp:20]: What is an equivalent for the test, and for `CPU_BIND_THREAD` properties?

Comment (vurusovs) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/infer_request/config.cpp:20]: I also just found that `InferRequestConfigTest` are migrated to separate `properties_tests.*pp`, so all is clear for me now. Thanks!

**Review by vurusovs — APPROVED** [review]

**Pull request merged**